Issue building - qt4 (QCryptographicHash::addData) - Linux

Hi,

Building Dwarf Therapist with qt4 fails with the error below. It seems to be unable to find function "QCryptographicHash::addData". It works correctly for qt5.

Build works again if commit 351db8d0522cd89825ae9dc50e19fa68cbe6df98.is reverted.

When building Dwarf Therapist with qt4 support instead of qt5 this error is received:

```
g++ -c -pipe -O2 -Wall -W -D_REENTRANT -DQT_NO_DEBUG -DQT_SCRIPT_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++ -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtScript -I/usr/include/qt4 -Iinc -Iinc/models -Iinc/grid_view -Iinc/docks -Iui -Ithirdparty/qtcolorpicker-2.6 -Ibin/release -Ibin/release -o bin/release/dfinstancelinux.o src/dfinstancelinux.cpp
src/dfinstancelinux.cpp: In member function ‘virtual QString DFInstanceLinux::calculate_checksum()’:
src/dfinstancelinux.cpp:100:63: error: no matching function for call to ‘QCryptographicHash::addData(QFile*)’
     if (!proc.open(QIODevice::ReadOnly) || !hash.addData(&proc)) {
                                                               ^
src/dfinstancelinux.cpp:100:63: note: candidates are:
In file included from /usr/include/qt4/QtCore/QtCore:107:0,
                 from inc/mainwindow.h:26,
                 from inc/dwarftherapist.h:33,
                 from inc/skill.h:30,
                 from inc/dwarf.h:28,
                 from inc/dfinstancelinux.h:27,
                 from src/dfinstancelinux.cpp:37:
/usr/include/qt4/QtCore/qcryptographichash.h:69:10: note: void QCryptographicHash::addData(const char*, int)
     void addData(const char *data, int length);
          ^
/usr/include/qt4/QtCore/qcryptographichash.h:69:10: note:   candidate expects 2 arguments, 1 provided
/usr/include/qt4/QtCore/qcryptographichash.h:70:10: note: void QCryptographicHash::addData(const QByteArray&)
     void addData(const QByteArray &data);
          ^
/usr/include/qt4/QtCore/qcryptographichash.h:70:10: note:   no known conversion for argument 1 from ‘QFile*’ to ‘const QByteArray&’
src/dfinstancelinux.cpp: In member function ‘virtual int DFInstanceLinux::read_raw(const VIRTADDR&, size_t, QByteArray&)’:
src/dfinstancelinux.cpp:231:38: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     remote_iov[0].iov_base = (void *)addr;
                                      ^
src/dfinstancelinux.cpp:239:98: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 4 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
             error(0, errno, "error reading %u bytes from 0x%0u to %p", bytes, addr, buffer.data());
                                                                                                  ^
src/dfinstancelinux.cpp: In member function ‘virtual int DFInstanceLinux::write_raw(const VIRTADDR&, const size_t&, void*)’:
src/dfinstancelinux.cpp:313:38: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     remote_iov[0].iov_base = (void *)addr;
                                      ^
src/dfinstancelinux.cpp: In member function ‘VIRTADDR DFInstanceLinux::alloc_chunk(int)’:
src/dfinstancelinux.cpp:696:41: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if ((m_alloc_end - m_alloc_start) < size) {
                                         ^
Makefile:1588: recipe for target 'bin/release/dfinstancelinux.o' failed
make: *** [bin/release/dfinstancelinux.o] Error 1
```

--- a/src/dfinstancelinux.cpp
+++ b/src/dfinstancelinux.cpp
@@ -97,10 +97,17 @@ QString DFInstanceLinux::calculate_checksum() {
     // ELF binaries don't seem to store a linker timestamp, so just MD5 the file.
     QFile proc(QString("/proc/%1/exe").arg(m_pid));
     QCryptographicHash hash(QCryptographicHash::Md5);
-    if (!proc.open(QIODevice::ReadOnly) || !hash.addData(&proc)) {
+    if (!proc.open(QIODevice::ReadOnly)
+#if QT_VERSION >= 0x050000
+        || !hash.addData(&proc)
+#endif
+        ) {
         LOGE << "FAILED TO READ DF EXECUTABLE";
         return QString("UNKNOWN");
     }
+#if QT_VERSION < 0x050000
+    hash.addData(proc.readAll());
+#endif
     QString md5 = hexify(hash.result().mid(0, 4)).toLower();
     TRACE << "GOT MD5:" << md5;
     return md5;

